dev/src/blocks/pc-custom-endpoint/edit.js

```diff
@@ -0,0 +1,194 @@
+	 */
+	const [ isPostChooserModalOpen, setIsPostChooserModalOpen ] =
+		useState( false );
+
+	const handleSelectPost = ( post ) => {
+		setAttributes( {
+			selectedPostID: post.ID, // Note: custom endpoint returns ID, not id
+			selectedPostTitle: post.post_title,
+			selectedPostExcerpt: post.post_excerpt,
+			selectedPostURL: post.guid,
+		} );
+		setIsPostChooserModalOpen( false );
+	};
+
+	const handleRemovePost = () => {
+		setAttributes( {
+			selectedPostID: undefined,
+			selectedPostTitle: undefined,
+			selectedPostExcerpt: undefined,
+			selectedPostURL: undefined,
+		} );
+	};
+
+	// Custom entity query parameters for the bob endpoint
+	const customEntityQuery = {
+		// The endpoint already filters by meta_endpoint_flag = 'yes'
+		// Add any additional parameters your endpoint might support
+		numberposts: 10,
+	};
+
+	return (
+		<div { ...useBlockProps() }>
+			<PostChooserSidebar
+				postID={ selectedPostID }
+				postTitle={ selectedPostTitle }
+				postURL={ selectedPostURL }
+				onRemovePost={ handleRemovePost }
+				openButtonLabel={ __( 'Select Bob Post' ) }
+				changeButtonLabel={ __( 'Change Bob Post' ) }
+				panelTitle={ __( 'Selected Bob Post' ) }
+				onOpenPostChooserModal={ () =>
+					setIsPostChooserModalOpen( true )
+				}
+			/>
+
+			<div className="wp-block-pc-custom-endpoint-container">
+				{ ! selectedPostID && (
+					<>
+						<p>
+							{ __(
+								'Custom endpoint post picker - no post selected.'
+							) }{ ' ' }
+							<Button
+								variant="link"
+								onClick={ () =>
+									setIsPostChooserModalOpen( true )
+								}
+							>
+								{ __( 'Select Bob Post' ) }
+							</Button>
+						</p>
+
+						{ isSelected && (
+							<Button
+								isPrimary
+								onClick={ () =>
+									setIsPostChooserModalOpen( true )
+								}
+							>
+								{ __( 'Select Bob Post' ) }
+							</Button>
+						) }
+					</>
+				) }
+
+				{ selectedPostID && (
+					<div className="selected-bob-post">
+						<h2 className="selected-bob-post-title">
+							<a href={ selectedPostURL }>
+								{ selectedPostTitle
+									? decodeEntities( selectedPostTitle )
+									: __( 'Untitled' ) }
+							</a>
+						</h2>
+
+						{ selectedPostExcerpt && (
+							<div
+								className="selected-bob-post-excerpt"
+								dangerouslySetInnerHTML={ {
+									__html: selectedPostExcerpt,
+								} }
+							/>
+						) }
+					</div>
+				) }
+
+				{ isPostChooserModalOpen && (
+					<PostChooser
+						modalLabel={ __(
+							'Search Bob posts with custom endpoint'
+						) }
+						searchPlaceholder={ __( 'Search Bob posts…' ) }
+						onSelectPost={ handleSelectPost }
+						modalTitle={ __( 'Choose a Bob Post' ) }
+						onClose={ () => setIsPostChooserModalOpen( false ) }
+						// Use the custom entity kind and name
+						entityKind="root"
+						postTypes={ [
+							{
+								label: __( 'Bob Posts' ),
+								value: 'block-imports/v1/bob-endpoint',
```

Comment:
I'm guessing this value here for the postType name is due to how the endpoint and custom entity is setup and named. If there is a reason to name it this way I'd like to understand it. If not, it might make it more readable and understandable to use the post type name here and in the last part of the endpoint path that identifies the resource being fetched. If this endpoint is getting the `import-bob` post type couldn't it just be `block-imports/v1/import-bob`? 

Or if this was a real world example: 
`bu.edu/admissions/wp-json/r-admissions/v1/courses` for a `courses` custom post type in Admissions theme/site.